### ISSUE TITLE
Choose Custom thumbnail image video upload 

### ISSUE THREAD

**Issue by RaviT-RSW** [Ziggeo/ZiggeoPhpSdk#34]
There is a feature to choose a video thumbnail image after recording video but there is no way to choose an image after video upload.
I have a requirement to choose an image in video upload. if this is already built can someone give example or reference?
Thank you

**Comment by Bane-D** [comment]
Hi there,

Since you posted on PHP SDK I would assume that you are looking into details from the server side.

There is a create video function: https://github.com/Ziggeo/ZiggeoPhpSdk#method-videos-create which is designed to make it easy for you to create a video. When using it, the image will be created by our servers for you, from the video that you have created.

If you want to create an image, you could do that by creating the video through few different steps instead.

First you will need to create the image, then the video and then bind them together.

Exact steps would be as follows:
1. Create empty object
2. Create stream
3. Attach cover shot
4. Attach video
5. Bind the stream

## 1. Create empty object

This is using the call to videos->create: https://github.com/Ziggeo/ZiggeoPhpSdk#method-videos-create however in this call you can provide any details except the video file itself.

Example:
```
$video_data = $ziggeo->videos()->create(array());
```

This creates what we call an empty object on our server, to which the video would be uploaded to.

Now you want to have that video token in use for the next calls (`$video_data['token']`).

## 2. Create Stream

We are using the video token and then calling the streams->create: https://github.com/Ziggeo/ZiggeoPhpSdk#create-1 however in this call you should also not provide the file information. This is to allow us to create the stream without kicking up the transcoding just yet.

We will however need to capture the stream info.

```
$stream_data = $ziggeo->streams()->create($video_data['token'], array()));
```
* Please note that $video_data['token'] requires us to have the data from videos->create call.

## Next steps

The next steps are simple, as you would just use the details above and make them in the order as mentioned here:
1. https://github.com/Ziggeo/ZiggeoPhpSdk#attach-image - to attach the image to the video and stream we created
2. https://github.com/Ziggeo/ZiggeoPhpSdk#attach-video - to attach the video to the video and stream we created
3. https://github.com/Ziggeo/ZiggeoPhpSdk#bind - to make the video actually finalized

By doing so, you should end up with the video that uses the image you have passed first.

Hope this helps, and do let us know how it goes :)

**Comment by RaviT-RSW** [comment]
Thank you for your reply and valuable time.
I tried this but get some uncaught errors
here is my code

```
           $tmp_arguments['file'] = $_FILES['files']['tmp_name'][0];
           $thumbnail['file'] = $_FILES['imagefiles']['tmp_name'][0];

           $video_data = $ziggeo->videos()->create(array());           ​
           ​$token = $video_data->token;

           ​$stream_data = $ziggeo->streams()->create($token, array());
           ​$ziggeo->streams()->attach_image($token, '$secrate_token', $thumbnail);
           ​$ziggeo->streams()->attach_video($token, '$secrate_token', $tmp_arguments);
           ​$ziggeo->streams()->bind($token, '$secrate_token');
```
I pass 2 arguments in a bind() because under ziggeo/classes/ZiggeoStreams.php   bind method has only 2 arguments as in the image

![bind](https://user-images.githubusercontent.com/88326465/149135215-f04519a6-53e0-4357-aa12-c8fccef41a1a.png)

Thank you again @Bane-D

**Comment by Bane-D** [comment]
Just quickly looking at the code you posted, I can not tell where the `$secrate_token` is from. This is just from a quick look and without knowing the error.

Will need to try and make a test on my side to be able to tell you more specific info about this.

**Comment by Bane-D** [comment]
Hi there,

I had checked the code needed for this yesterday and I noticed we are getting 412 - Precondition Failed error on bind call.

This error suggests that something is not passed through to our server, so I have reported this to my colleagues working on the API backend to check. Once we know more we will follow up with more details.

**Comment by Bane-D** [comment]
Hi,

Thank you for the wait. We have recently made some changes in regards to this and bind method is actually no longer needed.

So everything except the bind would work just fine. I have added a new demo that showcases this:
https://github.com/Ziggeo/ZiggeoPhpSdk/blob/master/demos/_streams_create_with_custom_image.php

You would call that demo as following in console:

```
php _streams_create_with_custom_image.php --app_token "APP_TOKEN" --private_key "PRIVATE_KEY" --path_image "../tests/assets/ziggeo-dashboard-php.png" --path_video "../tests/assets/video1s.mp4"
```

* You would replace "APP_TOKEN" and "PRIVATE_KEY" with your tokens and also the path with the ones you want to use.

Hope this helps, and do let us know how it goes.
